handbook.md

```diff
@@ -0,0 +1,256 @@
+---
+layout: contribute
+title: Handbook
+---
+
+# How we make things
+
+## Branching and SCM
+
+We use git for source control. We follow this model in what we call the core projects ([Foreman](https://github.com/theforeman/foreman), [Smart Proxy](https://github.com/theforeman/smart-proxy), [Foreman installer](https://github.com/theforeman/foreman-installer) and [Foreman SELinux](https://github.com/foreman-selinux)) for branching:
+
+* **develop**
+    * New pull requests should be made against this branch.
+    * All pull requests are merged here.
+    * Contains nightlies code. It's relatively stable, but we make no promises about it.
+    * [`test_develop` in Jenkins](http://ci.theforeman.org/job/test_develop/) ensures our tests are passing at all times for various configurations.
+* **release-stable (i.e. 1.10-stable)** -
+    * Individual developers do not need to maintain this, it's part of our [release process](http://projects.theforeman.org/projects/foreman/wiki/Release_Process#Preparing-a-release)
+    * Code is cherry-picked using `git cherry-pick -x` into this branch.
+    * It contains the latest stable code for a release cycle, 1.8-stable contains 1.8.1, 1.8.2, 1.8.3...
+    * Typically, we will announce the start of a new branch
+
+And we tag releases:
+
+* **release (i.e. 1.10)**
+    * Code is cherry-picked using `git cherry-pick -x` into this tag.
+    * It contains the latest stable code for a *specific release*, for instance, 1.10.0.
+
+We do not force push to any of these branches. If a commit must be reverted, use `git revert`, and push the reverted commit.
+
+Foreman projects strive to have a linear history ([like this](http://i.imgur.com/RMFS3bt.png)). A way to achieve this is to never create merge commits, by always rebasing your changes with develop `git pull --rebase upstream develop`. This allows us to easily revert our code to any point in time, which is helpful to detect in what commit we introduced an error using `git bisect`.
```

Comment:
+1 for leaving it in as-is. "Strives" is the keyword - we don't always manage it, but it's something I do on all my repos, and I think others should too.

side note: Clean log history has been a point in every talk on git I've ever been to (ok, that's like 3 talks, but still...), so I'm slightly surprised that GitHub *still* hasn't added --ff-only to it's merge button by default ;)